Dockerfile

```diff
@@ -76,9 +75,9 @@ WORKDIR /srv/sylius
 ARG APP_ENV=prod
 
 # prevent the reinstallation of vendors at every changes in the source code
-COPY composer.json composer.lock symfony.lock ./
```

Comment:
are you sure this should be removed?

Reply:
The composer.lock file is not needed anymore (since composer v2 as i remember?). The composer is creating it when building containers and modify directory on your machine (the composer.lock is created). I might be wrong with my thinking and the composer.lock file might be needed to prevent reinstalation.
When I checked manualy I made some changes in code of application, and run `docker-compose build` and the composer was running from cache ( no reinstallation of vendors happened).

Reply:
If somebody with docker experience is looking here please give an opinion, I am still learning docker 😄

Reply:
According to the composer docs here: https://getcomposer.org/doc/01-basic-usage.md#installing-with-composer-lock it is a good practice to commit your project's composer.lock file to version control. Also it makes sense to copy this file to Docker's context during image build, because composer then don't need to resolve the latest versions of a dependency but simply grabs the locked version of a dependency in the composer.lock file. This also ensures that you have a controlled Docker image build. Else it may happen that some dependencies changes during your Docker image build process which results in another behaviour.

So with the composer.lock file you can have controlled way of updating your dependencies, without it each `composer install` execution may install some other versions of a dependency which may produce some error in your application.

As far as I see this Sylius-Standard repo also does not include such a composer.lock file, so you need to make sure that you have run composer install (or composer create-project...) before you build a Docker image, else the Docker image build will fail with the message that "the composer.lock file does not exist...".

Reply:
Just a small question here, how would you run `composer install` if you rely on the PHP & Composer provided by the Docker image (which, as you said, will fail to built on first run b/c there is no `composer.lock`)?

Reply:
I would run it during local development, test the application and commit it to the repository. The CI/CD pipeline then simply uses the lock file for installing the dependencies, executing all the needed tests and finally builds the Docker image. It actually also depends how you would like to use the Docker container. I would strictly make here a separation between a "local development flow with Docker" and the Docker image build for a Production environment. The Docker image build for the Production environment should actually IMHO be built based on the lock file, else you can't be sure which dependency version finally get installed for your application. Also you should use the same PHP and composer version throughout the whole process in dev and Prod build.

If you would use the Docker image only or also for your local development flow then you really don't need to copy the lock file because then it is ok to simply have the composer.json and execute the composer install. But also there you should then commit the generated lock file to the repository to lock down/snapshot your current dependency version graph which works with your current application version.

Reply:
thanks @ilimination for your review comment, unfortunately I don't think that composer.lock file can be left. As in Sylius there are is a matrix of version supported (like php 7.4 or 8.0 and different mysql's) and leaving composer lock might interfere with it. Speaking about splitting prod and dev in two - well te bo honest I struggle with my experience at the moment to make it right 😄 .

Reply:
A fix for being able to build the Docker image before `composer install` (without an `composer.lock` file) was proposed here: https://github.com/Sylius/Sylius-Standard/pull/559

Reply:
Including `composer.lock` in the Sylius-Standard repo might indeed interfere with the various PHP versions, but only providing instructions to Docker to COPY your own `composer.lock` file once it's initialized should not interfere.

Reply:
I have added the line from #559 into the PR. Credits to you @tarnawski , hope this won't be a problem to use your code here? 😄

Reply:
A more proper way will be merge https://github.com/Sylius/Sylius-Standard/pull/559 and then rebase your changes to be aline with the master, but as is a small improvement - take it as yours :wink: